update flexs for source header and tabs (fixes #1489)

binarybaba opened the pull request:
Associated Issue: #1489 

### Summary of Changes

* toggle-button-start flexed 1
* toggle-button-end flexed 1
* source-header displayed flex and row wrapped
* new-tab btn now has 20px margin start

jasonLaster commented on src/components/SourceTabs.css:
```diff
@@ -134,6 +140,7 @@
 .toggle-button-start {
   top: 7px;
   left: 0;
+
```
Hmm, mind dropping the extra new lines

binarybaba replied:
Oh crap sorry i'll do that as soon as I reach home. Thanks!

jasonLaster commented:
Thanks @binarybaba! Just played with it and found a couple things

* the new tab button is slightly higher than the tabs
* the new tab button could use some hover styles (`cursor: pointer`, perhaps slightly darker)
* when the header is narrowed, the more tabs button overlays the new tab button

sorry if this is some new requirements that were unmentioned earlier

![](http://g.recordit.co/LjUTzJujad.gif)

binarybaba commented:
hokay I just came home a while ago and man I did not see that coming. I thought github would give me a space to write the summary but it didn't. Here's the stuff I changed:

- Changed the drop down button size to align more with the add new button
- Changed the toggle buttons alignment so they horizontally align better with right sidebar buttons
- Fixed the overlay problem
- Stylized the add new button a bit on hover
- Allowed overflow on source tabs hence when they're active, there is no bottom border.. so it looks more like a tab
- Removed the new lines finally haha

jasonLaster commented:
I dropped the new tab button styling from this PR because I'm not sure what treatment would be best.

Using the close button UI is not a bad idea though. Perhaps it's a little strong through.

binarybaba commented:
Hey it was your call in the end so kudos to you! I had real fun though. Thanks!

jasonLaster commented:
Likewise! 

Stick around. Stop by our office hours next week!

binarybaba commented:
Absolutely!
